cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/AbstractConnectOperator.java

```diff
@@ -610,21 +611,22 @@ private Future<ConnectorStatusAndConditions> autoRestartConnector(Reconciliation
             var count = autoRestartStatus.getCount();
             var minutesSinceLastRestart = StatusUtils.minutesDifferenceUntilNow(StatusUtils.isoUtcDatetime(autoRestartStatus.getLastRestartTimestamp()));
 
-            return count < 7 && minutesSinceLastRestart >= autoRestartBackOffInterval(count);
+            return (maxRestarts == null || count < maxRestarts)
+                    && minutesSinceLastRestart >= autoRestartBackOffInterval(count);
         }
     }
 
     /**
      * Calculates the back-off interval for auto-restarting the connectors. It is calculated as (n^2 + n) where n is the
-     * number of previous restarts. As a result, the restarts should be done after 0, 2, 6, 12, 20, 30 and 42 minutes
-     * (there are always only up to 7 restarts).
+     * number of previous restarts, but is capped at max 60 minutes. As a result, the restarts should be done after 0,
+     * 2, 6, 12, 20, 30, 42, and 56 minutes and then every 60 minutes.
      *
      * @param restartCount  Number of restarts already applied to the connector
      *
      * @return  Number of minutes after which the next restart should happen
      */
     private int autoRestartBackOffInterval(int restartCount)    {
```

Comment:
We rarely use minutes for things (usually ms or s), so the name of this method makes me a bit twitchy. Maybe we should take this opportunity to append `Mins` to the end of it?